spec/controllers/idv/verify_documents_controller_spec.rb

```diff
@@ -0,0 +1,358 @@
+require 'rails_helper'
+
+describe Idv::VerifyDocumentsController do
+  describe '#create' do
+    before do
+      sign_in_as_user
+      controller.current_user.document_capture_sessions.create!
+    end
+
+    subject(:action) { post :create, params: params }
+
+    let(:document_capture_session) { controller.current_user.document_capture_sessions.last }
+    let(:params) do
+      {
+        encryption_key: 'foo',
+        front_image_url: 'http://foo.com/bar1',
+        back_image_url: 'http://foo.com/bar2',
+        selfie_image_url: 'http://foo.com/bar3',
+        document_capture_session_uuid: document_capture_session.uuid,
+      }
+    end
+
+    context 'when document capture is not enabled' do
+      before do
+        allow(FeatureManagement).to receive(:document_capture_step_enabled?).and_return(false)
+      end
+
+      it 'disables the endpoint' do
+        action
+        expect(response).to be_not_found
+      end
+    end
+
+    context 'when document capture is enabled' do
+      before do
+        allow(FeatureManagement).to receive(:document_capture_step_enabled?).and_return(true)
+      end
+
+      context 'when fields are missing' do
+        before { params.delete(:front_image_url) }
+
+        it 'returns error status when not provided image fields' do
+          action
+
+          json = JSON.parse(response.body, symbolize_names: true)
+          expect(response.status).to eq(400)
+          expect(json[:success]).to eq(false)
+          expect(json[:errors]).to eq [
+            { field: 'front_image_url', message: I18n.t('doc_auth.errors.not_a_file') },
+          ]
+        end
+
+        it 'tracks analytics' do
+          stub_analytics
+
+          expect(@analytics).to receive(:track_event).with(
+            Analytics::IDV_DOC_AUTH_SUBMITTED_DOCUMENT_PROCESSING_FORM,
+            success: false,
+            errors: {
+              front_image_url: [I18n.t('doc_auth.errors.not_a_file')],
+            },
+            remaining_attempts: Figaro.env.acuant_max_attempts.to_i - 1,
+          )
+
+          expect(@analytics).not_to receive(:track_event).with(
+            Analytics::IDV_DOC_AUTH_SUBMITTED_IMAGE_UPLOAD_VENDOR,
+            any_args,
+          )
+
+          action
+        end
+      end
+
+      context 'when a value is not a url' do
+        before { params.merge!(front_image_url: 'some string') }
+
+        it 'returns an error' do
+          action
+
+          json = JSON.parse(response.body, symbolize_names: true)
+          expect(response.status).to eq(400)
+          expect(json[:errors]).to eq [
+            { field: 'front_image_url', message: I18n.t('doc_auth.errors.not_a_file') },
+          ]
+        end
+
+        context 'with a locale param' do
```

Comment:
WDYT of moving most of these specs to be unit test for the form?

Reply:
sure i can do that.  i was just following the format of the `images_upload_controller_spec.rb`.  seems like this type of spec would be more valuable.